src/python/gudhi/reader_utils.cc

```diff
@@ -0,0 +1,103 @@
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2020 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <gudhi/Points_off_io.h>
+#include <gudhi/reader_utils.h>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+
+#include <vector>
+#include <array>
+#include <string>
+
+namespace py = pybind11;
+
+
+PYBIND11_MODULE(reader_utils, m) {
+      m.attr("__license__") = "MIT";
+      m.def("read_points_from_off_file", [](const std::string& off_file) {
+            Gudhi::Points_off_reader<std::vector<double>> off_reader(off_file);
+            return off_reader.get_point_cloud();
+          },
+        py::arg("off_file"),
+        R"pbdoc(
+    Read points from OFF file.
+
+    :param off_file: An OFF file style name.
+    :type off_file: string
+
+    :returns:  The point set.
+    :rtype: List[List[float]]
+        )pbdoc"
+      );
+      m.def("read_lower_triangular_matrix_from_csv_file", &Gudhi::read_lower_triangular_matrix_from_csv_file<double>,
+        py::arg("csv_file"), py::arg("separator") = static_cast<const char>(';'),
+        R"pbdoc(
+    Read lower triangular matrix from a CSV style file.
+
+    :param csv_file: A CSV file style name.
+    :type csv_file: string
+    :param separator: The value separator in the CSV file. Default value is ';'
+    :type separator: char
+
+    :returns:  The lower triangular matrix.
+    :rtype: List[List[float]]
+        )pbdoc"
+      );
+      m.def("read_persistence_intervals_grouped_by_dimension", &Gudhi::read_persistence_intervals_grouped_by_dimension,
+        py::arg("persistence_file"),
+        R"pbdoc(
+    Reads a file containing persistence intervals.
+    Each line might contain 2, 3 or 4 values: [[field] dimension] birth death
+    The return value is a `dict(dim, list(tuple(birth, death)))`
+    where `dim` is an `int`, `birth` a `float`, and `death` a `float`.
+    Note: the function does not check that birth <= death.
+
+    :param persistence_file: A persistence file style name.
+    :type persistence_file: string
+
+    :returns:  The persistence pairs grouped by dimension.
+    :rtype: Dict[int, List[Tuple[float, float]]]
+        )pbdoc"
+      );
+      m.def("read_persistence_intervals_in_dimension", [](const std::string& persistence_file, int only_this_dim) {
+            // std::vector<std::pair<double, double>> as in read_persistence_intervals_in_dimension
+            // was not giving the good results because of memory alignment
+            // std::vector is mandatory as we don't know in advance the number of persistence intervals to be read
+            std::vector<std::array<double,2>> ret;
+            Gudhi::read_persistence_intervals_and_dimension(persistence_file,
```

Comment:
We can probably release the GIL during the C++ part of this function (it may need to be active again for the py::array part, but the slow part is the C++ I/O anyway).

Reply:
Done on 99ac0b4b